<h1>Bachelor of Science in Information Technology<h1>
<hr>
<h3>Student Analyst : Mark Dave L Rodriguez<h3>


In [56]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [57]:
penguins = pd.read_csv(r"C:\Users\CLIENT\Documents\penguin.csv")
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive analytics<h1>


In [58]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [59]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Cleaning Data<h1>

In [60]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [61]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [62]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [63]:
adelie_df = penguins[penguins['species']=="Adelie"]
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [64]:
chinstrap_df = penguins[penguins['species']=="Chinstrap"]
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [65]:
gentoo_df = penguins[penguins['species']=="Gentoo"]
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [66]:
penguins[penguins["body_mass_g"] >= 3000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
4       Adelie  Torgersen            36.7           19.3              193.0   
5       Adelie  Torgersen            39.3           20.6              190.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
5         3650.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[333 rows x 8 columns]

In [67]:
penguins[(penguins['species']=="Gentoo") & (penguins["body_mass_g"] >= 3000)]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
270  Gentoo  Biscoe            47.2           13.7              214.0   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
270       4925.0  female  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[123 rows x 8 columns]

In [68]:
penguins[(penguins['species']=="Gentoo") & (penguins["body_mass_g"] >= 3000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 123 entries, 152 to 275
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            123 non-null    object 
 1   island             123 non-null    object 
 2   bill_length_mm     123 non-null    float64
 3   bill_depth_mm      123 non-null    float64
 4   flipper_length_mm  123 non-null    float64
 5   body_mass_g        123 non-null    float64
 6   sex                119 non-null    object 
 7   year               123 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 8.6+ KB


In [69]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mode()[0])
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mode()[0])
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mode()[0])
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mode()[0])
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [70]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [71]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


<h1>DATA ANALYSIS<h1>

<h1>1. WHAT SPECIES HAS THE HIGHEST AVERAGE WEIGHT OF THEM ALL<h1>

In [72]:
gen_result = gentoo_df['body_mass_g'].mean()
print(f"the average weight of gentoo is : {gen_result}")


the average weight of gentoo is : 5076.016260162602


In [73]:
adel_result = adelie_df['body_mass_g'].mean()
print(f"the average weight of adelie is : {adel_result}")


the average weight of adelie is : 3700.662251655629


In [74]:
chin_result = chinstrap_df['body_mass_g'].mean()
print(f"the average weight of chinstrap is : {chin_result}")

the average weight of chinstrap is : 3733.0882352941176


<h3>INSIGHT : AS U CAN SEE FROM THE DATA COLLECTED THE HIGHEST AVERAGE WEIGHT GOES TO GENTOO WITH THE RESULT OF 5076.016260162602<h3>

<h1>2. WHAT SPECIES HAS THE LOWEST AVERAGE WEIGHT OF THEM ALL<h1>

In [75]:
gen_result = gentoo_df['body_mass_g'].mean()
print(f"the average weight of gentoo is : {gen_result}")

adel_result = adelie_df['body_mass_g'].mean()
print(f"the average weight of adelie is : {adel_result}")

chin_result = chinstrap_df['body_mass_g'].mean()
print(f"the average weight of chinstrap is : {chin_result}")



the average weight of gentoo is : 5076.016260162602
the average weight of adelie is : 3700.662251655629
the average weight of chinstrap is : 3733.0882352941176


<h3> INSIGHT : AS U CAN IN THE RESULT THE LOWEST AVERAGE WEIGHT IS ADELIE WITH RESULT OF 3700.662251655629<h3>

<h1>3. HOW MANY GENTOO DATA WAS RECORDED IN 2009<h1>

In [76]:
gentoo_df["year"].value_counts()

year
2008    46
2009    44
2007    34
Name: count, dtype: int64

<h3>INSIGHT : U CAN SEE THAT IN 2009 THEIRS 44 WAS COLLECTED IN THAT YEAR<h3>

<h1>4. WHAT IS THE TOTAL DATA COLLECTED IN 2007<h3>

In [77]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3> INSIGHTS : AS U CAN SEE THE DATA COLLECTED IN 2007 WAS 110<h3>

<h1>5. HOW MANY ADELIE IS ABOVE 4000G ?<h1>

In [78]:
adelie_df[adelie_df["body_mass_g"]>= 4000].info()

<class 'pandas.core.frame.DataFrame'>
Index: 39 entries, 7 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            39 non-null     object 
 1   island             39 non-null     object 
 2   bill_length_mm     39 non-null     float64
 3   bill_depth_mm      39 non-null     float64
 4   flipper_length_mm  39 non-null     float64
 5   body_mass_g        39 non-null     float64
 6   sex                38 non-null     object 
 7   year               39 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 2.7+ KB


<h3>INSIGHTS : OUT OF 151 DATA ENTRIES, ONLY 39 OF THEM WEIGH ABOVE 4000G.

<h1>6. HOW MANY MALE AND FEMALE PENGUINS ARE THEIR?<H1>

In [79]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3>INSIGHTS : THEIRS 179 MALE AND 165 FOR FEMALE

<H1>7. WHAT IS MOST COMMON ISLAND ?<H1>

In [80]:
penguins['island'].mode()

0    Biscoe
Name: island, dtype: category
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

<h3>INSIGHTS : THEIRS 3 COMMON ISLAND WHICH IS BISCOE , DREAM AND TORGERSEN<h3>

<H1>DATA AGGREGATION</H1>

<h3>How many species are there per island</h3>

In [81]:
penguins.groupby(['species','island']).count()

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  
species   island                             
Adelie    Biscoe              44   44    44  
          Dream               56   56    56  
          Torgersen           52   52    52  
Chinstrap Biscoe               0    0     0  
          Dream               68   68    68  
          Torgersen            0    0     0  
Gentoo    Biscoe             124  124   124  
          Dream                0    0     0  
          Torgersen            0    0     0

In [82]:
penguins.groupby(['species','island','sex']).size()

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h3>Show the collected data of all the species in each year </h3>

In [83]:
penguins.groupby(['species','year']).size()

species    year
Adelie     2007    50
           2008    50
           2009    52
Chinstrap  2007    26
           2008    18
           2009    24
Gentoo     2007    34
           2008    46
           2009    44
dtype: int64

In [84]:
penguins.groupby(adelie_df['year']).size()

year
2007.0    50
2008.0    50
2009.0    52
dtype: int64

In [85]:
penguins.groupby(['year','sex','island'])['body_mass_g'].mean()

year  sex     island   
2007  female  Biscoe       4345.238095
              Dream        3446.590909
              Torgersen    3475.000000
      male    Biscoe       5102.173913
              Dream        3902.083333
              Torgersen    3958.333333
2008  female  Biscoe       4225.806452
              Dream        3444.117647
              Torgersen    3518.750000
      male    Biscoe       5006.060606
              Dream        4114.705882
              Torgersen    4193.750000
2009  female  Biscoe       4403.571429
              Dream        3447.727273
              Torgersen    3193.750000
      male    Biscoe       5102.343750
              Dream        3935.227273
              Torgersen    3784.375000
Name: body_mass_g, dtype: float64

In [86]:
penguins.groupby(['species','sex','island'])['body_mass_g'].mean()

species    sex     island   
Adelie     female  Biscoe       3369.318182
                   Dream        3344.444444
                   Torgersen    3395.833333
           male    Biscoe       4050.000000
                   Dream        4008.620690
                   Torgersen    3975.892857
Chinstrap  female  Biscoe               NaN
                   Dream        3527.205882
                   Torgersen            NaN
           male    Biscoe               NaN
                   Dream        3938.970588
                   Torgersen            NaN
Gentoo     female  Biscoe       4679.741379
                   Dream                NaN
                   Torgersen            NaN
           male    Biscoe       5404.924242
                   Dream                NaN
                   Torgersen            NaN
Name: body_mass_g, dtype: float64

In [87]:
penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

np.float64(0.8713015663447136)

<h1>8. HOW MANY ADELIE PENGUINS WERE RECORDED IN 2007?<h1>

In [88]:

adel_record = penguins.groupby(['species', 'year']).size()['Adelie', 2007]
print(f"RECORDED ADELIE IN 2007 : {adel_record}")



RECORDED ADELIE IN 2007 : 50


<H1>9. SHOW THE AVERAGE NUMBER OF RECORDS PER SPECIES ACROSS ALL YEARS.<H1>

In [89]:
avg_spec = penguins.groupby('species').size().mean()
print(f"AVERAGE NUMBER OF SPECIES ACROSS ALL YEARS : {avg_spec}")


AVERAGE NUMBER OF SPECIES ACROSS ALL YEARS : 114.66666666666667


<h1>10.WHAT IS THE AVERAGE BODY MASS PER SPECIES?<h1>

In [90]:
penguins.groupby('species')['body_mass_g'].mean()


species
Adelie       3701.315789
Chinstrap    3733.088235
Gentoo       5065.725806
Name: body_mass_g, dtype: float64

<h1>11. AVERAGE BODY MASS PER YEAR<h1>

In [91]:
penguins.groupby('year')['body_mass_g'].mean()


year
2007    4121.590909
2008    4266.666667
2009    4206.875000
Name: body_mass_g, dtype: float64

<h1>12. HOW MANY PENGUINS WERE RECORDED ON EACH ISLAND PER YEAR?<h1>


In [92]:
penguins.groupby(['year', 'island']).size()


year  island   
2007  Biscoe       44
      Dream        46
      Torgersen    20
2008  Biscoe       64
      Dream        34
      Torgersen    16
2009  Biscoe       60
      Dream        44
      Torgersen    16
dtype: int64

<h1>13.HOW MANY CHINSTRAP PENGUINS WERE RECORDED IN 2007?<h1>

In [93]:
penguins.groupby(['species', 'year']).size()['Chinstrap', 2007]

np.int64(26)

<h1>14.HOW MANY GENTOO PENGUINS WERE RECORDED IN 2007?<h1>


In [94]:
penguins.groupby(['species', 'year']).size()['Gentoo', 2007]


np.int64(34)

<h1>15. What is the average body mass of each species per year?<h1>


In [95]:
penguins.groupby(['year', 'species'])['body_mass_g'].mean()


year  species  
2007  Adelie       3698.500000
      Chinstrap    3694.230769
      Gentoo       5070.588235
2008  Adelie       3742.000000
      Chinstrap    3800.000000
      Gentoo       5019.565217
2009  Adelie       3664.903846
      Chinstrap    3725.000000
      Gentoo       5110.227273
Name: body_mass_g, dtype: float64

<h1>16.What is the average body mass of penguins on Dream Island by species and sex?<h1>

In [96]:
penguins.groupby(['island', 'species', 'sex'])['body_mass_g'].mean()['Dream']


species    sex   
Adelie     female    3344.444444
           male      4008.620690
Chinstrap  female    3527.205882
           male      3938.970588
Gentoo     female            NaN
           male              NaN
Name: body_mass_g, dtype: float64

<h1>17. What is the average body mass of Adelie penguins by island and sex?<h1>

In [97]:
penguins.groupby(['species', 'island', 'sex'])['body_mass_g'].mean()['Adelie']


island     sex   
Biscoe     female    3369.318182
           male      4050.000000
Dream      female    3344.444444
           male      4008.620690
Torgersen  female    3395.833333
           male      3975.892857
Name: body_mass_g, dtype: float64

<h1>18. What is the average body mass of Chinstrap penguins by island and sex?<h1>

In [98]:
penguins.groupby(['species', 'island', 'sex'])['body_mass_g'].mean()['Chinstrap']

island     sex   
Biscoe     female            NaN
           male              NaN
Dream      female    3527.205882
           male      3938.970588
Torgersen  female            NaN
           male              NaN
Name: body_mass_g, dtype: float64

<h1>19. What is the average body mass of gentoo penguins by island and sex?<h1>

In [99]:
penguins.groupby(['species', 'island', 'sex'])['body_mass_g'].mean()['Gentoo']


island     sex   
Biscoe     female    4679.741379
           male      5404.924242
Dream      female            NaN
           male              NaN
Torgersen  female            NaN
           male              NaN
Name: body_mass_g, dtype: float64

<h1>20.How many Chinstrap penguins are recorded on each island by sex?<h1>


In [100]:
penguins.groupby(['species', 'island', 'sex']).size()['Chinstrap']


island     sex   
Biscoe     female     0
           male       0
Dream      female    34
           male      34
Torgersen  female     0
           male       0
dtype: int64